python/graphstorm/inference/np_infer.py

```diff
@@ -96,9 +97,15 @@ def infer(self, loader, save_embed_path, save_prediction_path=None,
             else:
                 embs = {ntype: embs}
         else:
-            embs = do_full_graph_inference(self._model, loader.data, fanout=loader.fanout,
+            if isinstance(self._model, GLEM):
+                embedding_model = self._model if self._model.inference_using_gnn else self._model.lm
+                decoding_model = self._model.gnn if self._model.inference_using_gnn else self._model.lm
```

Comment:
instead of changing the code here, can we have GLEM to return the right layers based on `inference_using_gnn`?

Reply:
basically, we need to define `node_input_encoder`, `gnn_encoder`, `decoder` in GLEM correctly.

Reply:
Addressed.